### PR TITLE
W-11464483: Raise Error usage within a reusable Mule Operation

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

File: modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/api/processor/MuleSdkRaiseErrorProcessor.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * Processor capable of raising errors within a Mule Operation's body on demand, given a type and optionally a message.
+ *
+ * @since 4.5
+ */
+public final class MuleSdkRaiseErrorProcessor extends AbstractRaiseErrorProcessor {
+
+  @Override
+  protected ComponentIdentifier calculateErrorIdentifier(String typeId) {
+    return builder().namespace("THIS").name(typeId).build();
```
Comment (asanguinetti) [inline modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/api/processor/MuleSdkRaiseErrorProcessor.java:23]: Have we checked what would happen if someone tries to do something like this?
`<operation:raise-error type="THIS:CUSTOM" />`
Or even:
`<operation:raise-error type="OTHER:CUSTOM" />`

Shouldn't we have a test for those cases showing the expected outcome (maybe error in both, or maybe ignoring the namespace in the first one)?

**Review by asanguinetti — CHANGES_REQUESTED** [review]

File: modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/api/processor/MuleSdkRaiseErrorProcessor.java (hunk 1 above)
Comment (asanguinetti) [reply modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/api/processor/MuleSdkRaiseErrorProcessor.java:23]: Also, we should probably extract the string "THIS" to a shared constant so can track its usages more easily.
Maybe making this one public or moving it to a constants class: https://github.com/mulesoft/mule/blob/eb7c7bb88896929865674e4f36aed563311ff565/modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkExtensionModelParser.java#L55

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Review by asanguinetti — APPROVED** [review]

**Pull request merged**